src/typedefs.jl

```diff
@@ -19,7 +19,7 @@ It allows users to easily define operations on arrays with null values by
 reusing operations that only work on arrays without any null values.
 """
 immutable NullableArray{T, N} <: AbstractArray{Nullable{T}, N}
-    values::Array{T, N}
+    values::AbstractArray{T, N}
```

Comment:
This should be made a type parameter of the array or it's going to kill performance.